core/src/lib/region_locking.h

```diff
@@ -0,0 +1,12 @@
+#ifndef REGION_LOCKING_H
```

Comment:
missing license header, may require mentioning of the original author if the code was copied/imported